core/src/main/java/tc/oc/pgm/rotation/MapPoolManager.java

```diff
@@ -264,6 +264,14 @@ public void matchEnded(Match match) {
     activeMapPool.matchEnded(match);
   }
 
+  @Override
+  public Duration getCycleTime() {
+    if (activeMapPool != null && !activeMapPool.getCycleTime().isNegative()) {
+      return activeMapPool.getCycleTime();
+    }
+    return PGM.get().getConfiguration().getCycleTime();
```

Comment:
```suggestion
    Duration cycleTime;
    if (activeMapPool != null && !(cycleTime = activeMapPool.getCycleTime()).isNegative()) {
      return cycleTime;
    }
    return PGM.get().getConfiguration().getCycleTime();
```

Just avoiding the extra call to get cycle time

Reply:
Thanks @Pablete1234, adjusted 👍